kani-compiler/src/codegen_cprover_gotoc/codegen/contract.rs

```diff
@@ -0,0 +1,100 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+//! This file contains functions related to codegenning code contracts from MIR into gotoc
+use crate::codegen_cprover_gotoc::GotocCtx;
+use cbmc::goto_program::{Contract, Expr, Spec, Symbol};
+use rustc_ast::NestedMetaItem;
+use rustc_middle::mir::Local;
+use rustc_middle::ty::{self, FnSig};
+
+impl<'tcx> GotocCtx<'tcx> {
+    /// Transforms current function's arguments into expressions
+    fn extract_function_arguments(&mut self, sig: FnSig<'tcx>) -> Vec<Expr> {
```

Comment:
This function and the one bellow should be inside functions.rs, but since this is going into a branch for now, it might make sense to keep them here.

Another thing you could do is to create a PR to main that creates these functions inside function.rs and you just use them here. That will reduce the amount of divergence between the feature branch and main.

That said, not a blocker to this PR.